### ISSUE TITLE
Unable to start NzbGet using NAS Diskstation DS413 running DSM 6.x 

### ISSUE THREAD

**Issue by ccwillia** [nzbgetcom/nzbget#467]
### Is there already an issue for your problem?

- [X] I have checked older issues, open and closed

### NZBGet Version

v24.5-stable

### Platform

NAS/Synology/QNAP

### Environment

```markdown
NAS Diskstation DS413 running DSM 6.x
```

### Current Behavior

Upgraded from 24.1 to 24.5 which will not start.  Logs mention lscpu not found and unable to determine cpu.  I have tried to uninstall and reinstall but Nzbget will not run****

### Expected Behavior

Nzbget would run after upgrade to 24.5

### Steps To Reproduce

Use DSM community to install nzbGet
NzbGet never starts

### Logs

Starting nzbget command /volume1/@appstore/nzbget/bin/nzbget -c /volume1/@appstore/nzbget/var/nzbget.conf -o WebDir=/volume1/@appstore/nzbget/bin/webui -o LockFile=/volume1/@appstore/nzbget/var/nzbget.pid -D
sh: lscpu: command not found
Thu Dec 12 22:43:38 2024    DETAIL  Failed to get CPU model from '/proc/cpuinfo'
Thu Dec 12 22:43:38 2024    DETAIL  Failed to get CPU model from 'lscpu'
Thu Dec 12 22:43:38 2024    INFO    nzbget 24.5 daemon-mode
Thu Dec 12 22:43:38 2024    INFO    using /volume1/@appstore/nzbget/var/nzbget.conf
Thu Dec 12 22:43:38 2024    INFO    nzbget runs on 0.0.0.0:6789

**Comment by ccwillia** [comment]
Is it possible to move back to the 24.1 version until the issue is fixed?  I wasn't able to find a noarch 24.1 version to install manually.  Thanks in advance!

**Comment by dnzbk** [comment]
Yes, you can find and download previous versions [here](https://github.com/nzbgetcom/nzbget/releases).

What’s the processor model in your Synology??

**Comment by phnzb** [comment]
@ccwillia Through the SynoCommunity package you can only install the latest testing / latest stable version, unfortunately downgrading is not supported.
The old version of SPKs doesn't support DSM6, so that's not possible either.
You have 2 options:
- install the old nzbget.net version via https://synocommunity.com/package/nzbget - version 20220406-29 (https://packages.synocommunity.com/nzbget/29/nzbget.v29.f7321%5Bnoarch%5D.spk).
- downgrade nzbget manually via ssh terminal:
1. Make sure SSH service is enabled (Control Panel - Terminal & SNMP - Enable SSH service)
2. Make sure User home service is enabled (Control Panel - User - Advanced - Enable user home service)
3. Install nzbget via Package Center (latest stable version)
4. Stop nzbget service via Package Center.
5. Find the link to the version of the linux installer you want to downgrade to (in example I will use 24.1 - https://github.com/nzbgetcom/nzbget/releases/download/v24.1/nzbget-24.1-bin-linux.run)
6. Login to NAS via SSH
7. Run from terminal (use link from p.5, for sudo prompt use yours user password):
```
curl -LO https://github.com/nzbgetcom/nzbget/releases/download/v24.1/nzbget-24.1-bin-linux.run
sh nzbget-24.1-bin-linux.run
sudo cp nzbget/nzbget /volume1/@appstore/nzbget/bin/nzbget
sudo cp nzbget/7za /volume1/@appstore/nzbget/bin/7za
sudo cp nzbget/unrar /volume1/@appstore/nzbget/bin/unrar
sudo cp -r nzbget/webui /volume1/@appstore/nzbget/bin
sudo chown sc-nzbget:sc-download /volume1/@appstore/nzbget/bin/*
rm -rf ./nzbget
rm nzbget-24.1-bin-linux.run
```
9. Start nzbget via Package Center and do not upgrade it via internal updater

**Comment by ccwillia** [comment]
I am able to install and run the old nzbget.net version 21.1.  I will try updating again through DSM then run the steps you provided below to move back to 24.1 which was working prior to clicking the internal updater.  Hopefully this issue in 24.5 or future release can be fixed so i can keep current with the releases

**Comment by ccwillia** [comment]
Thanks for the steps above i was able to move back to 24.1 which runs successfully.  Let me know if any additional information is need to address the bug or if there is a future fix you would like me to test

**Comment by dnzbk** [comment]
@ccwillia Happy holidays!
I made a test build for you. 
Could you test it if it runs properly on your Synology?

[nzbget-ppc500-testing.tar.gz](https://github.com/user-attachments/files/18260909/nzbget-ppc500-testing.tar.gz)

Just extract it and launch:
`$ nzbget -s`

Thank you.

**Comment by ccwillia** [comment]
Thanks, I will give it a try soon

**Comment by ccwillia** [comment]
When trying to run i get the following error.  I did stop the nzbget client through the DSM interface before trying to run the test executable provided.  

`ERROR   Binding socket failed for 0.0.0.0: Error 38 - Function not implemented`

**Comment by dnzbk** [comment]
Could you please provide the output of `uname -a` and the `nzbget.log file`? 
Please the logs for any personal info before sharing.
I built the app targeting Linux kernel 3.2.40. It is possible that your Linux version is even older.

**Comment by ccwillia** [comment]
Here is the output from `uname -a`

`Linux DiskStation 2.6.32.12 #25556 SMP Fri Aug 16 14:56:56 CST 2024 ppc GNU/Linux synology_qoriq_413`

I do not see a nzbget.log file in the `nzbget-ppc500-testing` directory where i ran `./nzbget -s`

**Comment by dnzbk** [comment]
`2.6.32.12` - this is a very old version of the kernel. Is this the latest version of DSM available for your system? 
I’ll try to build the app for that version.

**Comment by ccwillia** [comment]
Yep, i am on version DSM 6.2.4-25556 Update 8 with no option to update any further

**Comment by ccwillia** [comment]
I checked with my brother who also runs an older Diskstation.  His Diskstation is running a similar version but has a different architecture and NzbGet 24.5 runs fine on his.  So it appears to be related to the architecture  

`Linux DiskStation 2.6.32.12 #25556 Thu Jul 1 14:25:48 CST 2021 armv5tel GNU/Linux synology_88f6281_211j`

**Comment by dnzbk** [comment]
We use the` uClibc` library for PowerPC x32 and `musl` for other architectures. This is because `musl` does not support PowerPC x32. As a result, we may observe differences in behavior.
Anyway, I managed to build the app for your Linux version:
[nzbget-ppc500-testing_2.tar.gz](https://github.com/user-attachments/files/18305815/nzbget-ppc500-testing_2.tar.gz)
Please test it out, thank you.

**Comment by ccwillia** [comment]
This version works.  For now i have replaced my old executable with this new one.

**Comment by ccwillia** [comment]
Well i thought everything was working but i am getting the following errors when processing nzb files.  There is over 1TB of space available on /volume1

`Sun Jan  5 16:44:37 2025    ERROR   Could not create file /volume1/downloads/incomplete/intermediate/***.#1/3.out.tmp: File too large`

**Comment by dnzbk** [comment]
I'm not sure if that is app related. Does 24.1 work fine with the same setup and NZB?
Could you please show the output of `lsblk -f`?

**Comment by ccwillia** [comment]
The `lsblk` command is not found, but i removed the intermediate directory in the paths configuration and it seems to be working correctly now.  Maybe it was a permissions problem?  I will test with some more files

**Comment by ccwillia** [comment]
Everything has been working.  So the fixes you have made should be good to go.  I appreciate the time you have spent to address the issues for my Diskstation!  Can this be merged into the larger code base so i can use the software update option within Nzbget in the future?

**Comment by dnzbk** [comment]
Great, I’ll merge my fixes into the codebase. I was waiting for your confirmation that everything is working fine. 
Thank you for testing.

**Comment by ccwillia** [comment]
Sounds good, do you want me to close this issue or will you be closing it once code has been merged?

**Comment by dnzbk** [comment]
Yes, I’ll close it myself when I merge my fixes, thanks.
